modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -99,6 +99,37 @@ class PandasDataframePartitionManager(ClassLogger, ABC):
     _column_partitions_class = None
     # Row partitions class is the class to use to create the row partitions.
     _row_partition_class = None
+    _execution_wrapper = None
+
+    @classmethod
+    def materialize_futures(cls, input_list):
+        """
+        Materialize all futures in the input list.
+
+        Parameters
+        ----------
+        input_list : list
+            The list that has to be manipulated.
+
+        Returns
+        -------
+        list
+            Modified input list.
+        """
+        # Do nothing if input_list is None or [].
+        if not input_list:
+            return input_list
```

Comment:
Otherwise, a reference is returned, using which one can accidentally change the input object.

```suggestion
            return []
```